src/basic/list.h

```diff
@@ -181,3 +181,12 @@
                 }                                                       \
                 (b) = NULL;                                             \
         } while (false)
+
+#define LIST_POP(name, a)                                               \
+        ({                                                              \
+                typeof(a)* _a = &(a);                                   \
+                typeof(a) _p = *_a;                                     \
```

Comment:
Is this done to avoid the double evaluation?

Reply:
yes, precisely, it's a common pattern in macros like this